actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -435,7 +435,8 @@ def form_for(record, options = {}, &block)
           object      = nil
         else
           object      = record.is_a?(Array) ? record.last : record
-          raise ArgumentError, "First argument in form cannot contain nil or be empty" unless object
+          contains_nil = record.nil? || (record.is_a?(Array) && record.include?(nil))
```

Comment:
Also jut wondering out loud but this looks like a breaking change, though it is correct.